docs/source/python/integration/python_r.rst

```diff
@@ -0,0 +1,312 @@
+    with localconverter(arrowconverter):
+        r_result = addthree(array)
+
+    # The result of the R function will be an R Environment
+    # we can convert the Environment back to a pyarrow Array
+    # using the rarrow_to_py_array function
+    py_result = rarrow_to_py_array(r_result)
+    print("RESULT", type(py_result), py_result)
+
+Running the newly modified ``addthree.py`` should now properly execute
+the R function and print the resulting PyArrow Array:
+
+.. code-block:: bash
+
+    $ python addthree.py
+    RESULT <class 'pyarrow.lib.Int64Array'> [
+      4,
+      5,
+      6
+    ]
+
+For additional information you can refer to
+`rpy2 Documentation <https://rpy2.github.io/doc/latest/html/index.html>`_
+and `rpy2-arrow Documentation <https://rpy2.github.io/rpy2-arrow/version/main/html/index.html>`_
+
+Invoking Python functions from R
+--------------------------------
+
+Exposing Python functions to R can be done through the ``reticulate``
+library. For example if we want to invoke :func:`pyarrow.compute.add` from
+R on an Array created in R we can do so importing ``pyarrow`` in R
+through ``reticulate``.
+
+A basic ``addthree.R`` script that invokes ``add`` to add ``3`` to
+an R array would look like:
+
+.. code-block:: R
+
+    # Load arrow and reticulate libraries
+    library(arrow)
+    library(reticulate)
+
+    # Create a new array in R
+    a <- Array$create(c(1, 2, 3))
+
+    # Make pyarrow.compute available to R
+    pc <- import("pyarrow.compute")
+
+    # Invoke pyarrow.compute.add with the array and 3
+    # This will add 3 to all elements of the array and return a new Array
+    result <- pc$add(a, 3)
+
+    # Print the result to confirm it's what we expect
+    print(result)
+
+Invoking the ``addthree.R`` script will print the outcome of adding
+``3`` to all the elements of the original ``Array$create(c(1, 2, 3))`` array:
+
+.. code-block:: bash
+
+    $ R --silent -f addthree.R 
+    Array
+    <double>
+    [
+      4,
+      5,
+      6
+    ]
+
+For additional information you can refer to
+`Reticulate Documentation <https://rstudio.github.io/reticulate/>`_
+and to the `R Arrow documentation <https://arrow.apache.org/docs/r/articles/python.html#using>`_
+
+R to Python communication using C Data Interface
+------------------------------------------------
+
+Both the solutions described in previous chapters use the Arrow C Data
+interface under the hood.
+
+In case we want to extend the previous ``addthree`` example to switch
+from using ``rpy2-arrow`` to using the plain C Data interface we can
+do so by introducing some modifications to our codebase.
+
+To enable importing the Arrow Array from the C Data interface we have to
+wrap our ``addthree`` function in a function that does the extra work
+necessary to import an Arrow Array in R from the C Data interface.
+
+That work will be done by the ``addthree_cdata`` function which invokes the
+``addthree`` function once the Array is imported.
```

Comment:
This entire snippet doesn't seem very informative to me ("introduce some modifications"... "do the extra work"...). Perhaps condense them into one or two meaningful sentences?

Also, this does not explain _why_ I would want to avoid ``rpy2-arrow`` and instead invoke the C Data Interface directly.

Reply:
@amol- please address this in a follow-up.